src/main/java/com/netgrif/application/engine/workflow/service/DataService.java

```diff
@@ -631,10 +632,7 @@ public boolean deleteFile(String taskId, String fieldId) {
             }
             useCase.getDataSet().get(field.getStringId()).setValue(null);
         }
-
-        updateDataset(useCase);
-        workflowService.save(useCase);
```

Comment:
Is this correct? If the case is not saved how the new state of the case's dataset is propagated?

Reply:
Yes, it is correct,  on the line below `return new SetDataEventOutcome(useCase, task, getChangedFieldByFileFieldContainer(fieldId, task, useCase));` the function call `getChangedFieldByFileFieldContainer` contains these removed lines.